controllers/progressiverollout_controller_unit_test.go

```diff
@@ -0,0 +1,35 @@
+package controllers
```

Comment:
Question for the wider audience: do we want to have a separate test file for the unit tests, or can we merge those tests with `progressiverollout_controller_test.go`?

I would prefer a single test file for simplicity, but I'm open to suggestions

Reply:
There's an argument to separate integration tests vs unit tests as they look and are different.
How is it done on other operators?

Reply:
I have never seen anything other controller.go and controller_test.go but I am happy if anyone can findaa different example.

Reply:
In other languages (such as Python or Java), tests would be in their own folder, and I feel like I'd separate unit and other types of tests in different files too. So I felt this makes sense.

But Golang seems to be different in that tests generally live in the same package as the code under test; so perhaps a 1:1 relation between code-under-test and test-code files is ideal? If so, I can move them to the same file. I'm not opinionated on this.